Domains/4-Application/DrawingProduction/SheetNumberingViews.ecschema.xml

```diff
@@ -0,0 +1,184 @@
+    </ECEntityClass>
+    <ECEntityClass typeName="SheetIndexView" modifier="Abstract" displayLabel="SheetIndex View" description="View that returns the total number of sheets in a sheet index">
+        <ECCustomAttributes>
+            <QueryView xmlns="ECDbMap.02.00.04">
+                <Query>
+                    SELECT 
+                        ECInstanceId,
+                        ECClassId,
+                        PageCount
+                    FROM (
+                        WITH
+                        -- General Notes:
+                        -- There are a number of times that column names are aliased arbitrarily.
+                        --  This is to avoid an issue with CTEs not respecting qualifiers.  See this issue: https://github.com/iTwin/itwinjs-core/issues/8571
+
+                        -- Recursively traverses the tree from root to leaves, building hierarchical paths.
+                        treeTraversal ([ECInstanceId], [ECClassId], [DirectParentId]) AS (
+                        -- Start with root nodes-SheetIndexes
+                        SELECT
+                            si.ECInstanceId,
+                            si.ECClassId,
+                            NULL AS [DirectParentId]
+                        FROM bis:SheetIndex si
+
+                        UNION ALL
+                            -- Recursively add children to parents, building hierarchical sort paths
+                            SELECT
+                            child.ECInstanceId,
+                            child.ECClassId,
+                            child.NodeParentId AS [DirectParentId]
+                            FROM treeTraversal parent
+                        JOIN (
+                        -- A table of all possible children joined to their parents
+                                SELECT
+                                    [sr].[ECInstanceId],
+                                    ECClassId,
+                                    [sr].[Parent].[Id] AS [NodeParentId]
+                                FROM
+                                    [bis].[SheetReference] [sr]
+
+                                UNION ALL
+
+                                SELECT
+                                    [sif].[ECInstanceId],
+                                    ECClassId,
+                                    [sif].[Parent].[Id] AS [NodeParentId]
+                                FROM
+                                    [bis].[SheetIndexFolder] [sif]
+                            ) child ON child.NodeParentId = parent.ECInstanceId
+                        )
+                        -- Final result: Returns only SheetReference node due to inner join
+                        SELECT
+                            sr.ECInstanceId AS [ECInstanceId],
+                            ec_classid('SheetNumbering', 'SheetIndexView') [ECClassId],
```

Comment:
Since this view conceptually extends the persisted definition of a bis.SheetIndex, it should return its ECClassId (i.e. ec_classid('BisCore', 'SheetIndex') ) rather than the View-class' ECClassId.

Reply:
Done